sections/appendix/gibbs_sampling.tex

```diff
@@ -0,0 +1,82 @@
+\subsection{Gibbs Sampling}\label{sec:appendix_gibbs}
+The Gibbs Sampling algorithm consists of two procedures: Random Initialization and Gibbs Sampling.
+In the following sections, we explain how these procedures have been implemented.
+
+\subsubsection{Random Initialization}
+Before a Gibbs sampling algorithm can run, word occurrences need to be randomly initialized with topics.
+To do this, we iterate over each word within each document and assign a random topic for it.
+This procedure is shown in \autoref{lst:randomInit}.
+\begin{lstlisting}[language=Python, caption=Random Initialization,label={lst:randomInit}, float, floatplacement=H]
+import numpy as np
+
+def rand_initialize(documents: List[np.ndarray]):
+	wt_assignment = []
+	for doc in documents:
+		curr_doc = []
+		for word in doc:
+			# Construct the topic distribution
+			pz = _conditional_distribution()
+			
+			# Draw a new topic and assign it
+			t = np.random.multinomial(1, pz).argmax()
+			curr_doc.append(t)
+			
+			# Increase the topic counts
+			increase_count()
+		wt_assignment.append(curr_doc)
+	return wt_assignment
+\end{lstlisting}
+
+Various parameters are left out to simplify the code listing of our initialization method.
+The \emph{\_conditional\_distribution()} function creates a distribution of topics based on the current word.
+It is based on \autoref{eq:gibbs_eq}~\cite{author_topic_2012}.
+\begin{equation}\label{eq:gibbs_eq}
+	\begin{split}
+		P(z_i = k|w_i = m, \boldsymbol{z}_{-i}, \boldsymbol{w}_{-i}, \alpha, \beta) = 
+		\underbrace{\frac{C^{DT}_{dk} + \alpha}{\sum_{k'} C^{DT}_{dk'} + T\alpha}}_{Doc-Topic}
+		\underbrace{\frac{C^{WT}_{mk} + \beta}{\sum_{m'} C^{WT}_{m'k} + V\beta}}_{Topic-Word}
+	\end{split}
+\end{equation}
+where $C^{DT}_{dk}$ is the number of times document $d$ uses topic $k$.
+
+$C^{WT}_{mk}$ is the number of times topic $k$ uses word $m$.
+$\boldsymbol{z}_{-i}$ represents the topic assignments where the current instance is disregarded.
+$\alpha$ and $\beta$ is the Dirichlet parameter for the document-topic and topic-word distribution, respectively.
+The first fraction describes how likely topic $t$ appearing in document $d$ is, and the second fraction describes which words are most probable in topic $t$.
+Following the code in \autoref{lst:randomInit}, as we initialize, the words get higher probability of clustering together, since we update the counts every time, we look at a new word in line 16.
```

Comment:
```suggestion
Following the code in \autoref{lst:randomInit}, as we initialize, the words get a higher probability of clustering together, since we update the counts every time, we look at a new word in line 16.
```